Bio/SeqIO/Interfaces.py

```diff
@@ -58,21 +58,19 @@ def __init__(self, source, alphabet=None, mode="t", fmt=None):
             else:
                 raise ValueError(f"Unknown mode '{mode}'") from None
             self.stream = source
-            self.should_close_stream = False
+            # self.should_close_stream = False
         try:
             self.records = self.parse(self.stream)
         except Exception:
-            if self.should_close_stream:
-                self.stream.close()
+            self.stream.close()
             raise
 
     def __next__(self):
         """Return the next entry."""
         try:
             return next(self.records)
         except Exception:
-            if self.should_close_stream:
-                self.stream.close()
+            self.stream.close()
```

Comment:
I think this should be conditional, based on your ``__exit__`` code perhaps:

```
if self.stream is not self.source:
    self.stream.close()
```

But maybe you shouldn't close at all if there is an exception - would ``__exit__`` not get called after this?